logic/overworld.py

```diff
@@ -393,59 +395,62 @@ def __init__(self, options, world_setup, r):
 
         if options.logic == 'hell':
-            swamp.connect(forest, None) # damage boost from toadstool area across the pit.
-            forest_heartpiece.connect(forest, PEGASUS_BOOTS, one_way=True) # boots bonk across the pits
+            swamp.connect(forest_toadstool, SHIELD) # damage boost from toadstool area across the pit.
+            forest.connect(forest_heartpiece, PEGASUS_BOOTS, one_way=True) # boots bonk across the pits
             log_cave_heartpiece.connect(forest_cave, BOOMERANG) # clip the boomerang through the corner gaps on top right to grab the item
             writes_hut_outside.connect(swamp, PEGASUS_BOOTS) # boots bonk telephone booth
-        #     writes_cave_left_chest.connect(writes_hut, None) # damage boost off the zol to get across the pit. added bush since a requirement is necessary
+            writes_cave_left_chest.connect(writes_cave, None) # damage boost off the zol to get across the pit.
             graveyard.connect(forest, OR(PEGASUS_BOOTS, HOOKSHOT)) # boots bonk witches hut, or hookshot spam across the pit
             dungeon3_entrance.connect(ukuku_prairie, PEGASUS_BOOTS) # boots bonk + water buffer across bottom wall all the way to the entrance
-        #     prairie_3gap_stairs.connect(center_area, PEGASUS_BOOTS) # pit buffer to clip bottom wall and boots bonk across
-        #     prairie_plateau.connect(center_area, AND(BOMB, OR(PEGASUS_BOOTS, HOOKSHOT))) # boots bonk across pits, or hookshot spam
-        #     dungeon5_entrance.connect(center_area, FEATHER) # jesus jump into d5 entrance (wall clip or fast fall), wall clip to get out
-        #     richard_cave_chest.connect(richard_cave, PEGASUS_BOOTS) # boots bonk
-        #     #castle.connect(center_area, AND(PEGASUS_BOOTS, MEDICINE, OR(BOMB, BOOMERANG, MAGIC_POWDER, MAGIC_ROD, SWORD))) # medicine iframe abuse to get across spikes
-        #     into_to_mountains.connect(below_mountains, OR(HOOKSHOT, AND(FEATHER, PEGASUS_BOOTS, OR(SWORD, MAGIC_ROD, BOW)))) # get past crystal rocks by hookshotting into top pushable block, or boots dashing into top wall where the pushable block is
-        #     face_shrine.connect(dungeon6_entrance, PEGASUS_BOOTS, one_way=True) # jesus jump from top to island with quick falling on top, boots bonk across water and pits
-        #     bridge_seashell.connect(right_mountains_2, AND(PEGASUS_BOOTS, POWER_BRACELET)) # boots bonk
+            self._addEntranceRequirement("castle_jump_cave", PEGASUS_BOOTS) # 1 pit buffer to clip bottom wall and jump across.
+            prairie_cave_secret_exit.connect(prairie_cave, AND(BOMB, HOOKSHOT)) # hookshot spam across pits
+            prairie_cave.connect(prairie_cave_secret_exit, AND(BOMB, PEGASUS_BOOTS), one_way=True) # boots bonk across pits, can only go one way because of pits at bomb wall
+            ukuku_prairie.connect(d5_entrance, FEATHER) # jesus jump into d5 entrance (wall clip or fast fall), wall clip to get out

```

Comment:
shouldn't this be one way? As you cannot get from the D5 entrance to ukuku prarie? (Not that it really matters for the current randomization process, but it's nice to have things correct.

Reply:
yea indeed, I forgot the way out would lead to lower bay area instead of ukuku prairie